### PR TITLE
Add support for poolConfig in Salesforce connector

### PR DESCRIPTION
**Pull request by aashikam**
# Description
fixes: https://github.com/wso2-enterprise/internal-support-ballerina/issues/815

### CODE REVIEW

File: ballerina/types.bal [hunk 1]
```diff
@@ -69,12 +69,62 @@ public enum ColumnDelimiterEnum {
...
-    *config:ConnectionConfig;
     # The Salesforce endpoint URL
     string baseUrl;
     # Configurations related to client authentication
     http:BearerTokenConfig|config:OAuth2RefreshTokenGrantConfig|
         config:OAuth2PasswordGrantConfig|config:OAuth2ClientCredentialsGrantConfig auth;
+    # The HTTP version understood by the client
+    http:HttpVersion httpVersion = http:HTTP_2_0;
+    # Configurations related to HTTP/1.x protocol
+    ClientHttp1Settings http1Settings?;
```
Comment (ayeshLK) [inline ballerina/types.bal:80]: Any reason not to use `http:ClientHttp1Settings` [1] config here ? 

[1] - https://central.ballerina.io/ballerina/http/2.12.2#ClientHttp1Settings

File: ballerina/types.bal [hunk 2]
```diff
@@ -69,12 +69,62 @@ public enum ColumnDelimiterEnum {
...
+|};
+
+# Provides settings related to HTTP/1.x protocol.
+public type ClientHttp1Settings record {|
+    # Specifies whether to reuse a connection for multiple requests
+    http:KeepAlive keepAlive = http:KEEPALIVE_AUTO;
+    # The chunking behaviour of the request
+    http:Chunking chunking = http:CHUNKING_AUTO;
+    # Proxy server related options
+    ProxyConfig proxy?;
```
Comment (ayeshLK) [inline ballerina/types.bal:114]: Can't we use the `http:ProxyConfig` [1] here ? The record definition looks the same.

[1] -https://central.ballerina.io/ballerina/http/2.12.2#ProxyConfig

File: ballerina/Ballerina.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 distribution = "2201.10.0"
 org = "ballerinax"
 name = "salesforce"
-version = "8.1.0"
```
Comment (ayeshLK) [inline ballerina/Ballerina.toml:5]: The change is more likely a breaking change (as we are adding new fields to a closed record), hence better to check the possibility of releasing a major version or a minor-version with a new distribution update version.

**Comment by aashikam** [comment]
@ayeshLK 

https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/ballerina/types.bal We put the client init as same as the generated clients. 

Major version release is not possible at the moment with how urgent this fix is needed. (There are lot more to be fixed with the next major release: https://github.com/ballerina-platform/ballerina-library/issues/6697)
Anyway this is up for debate whether or not we are doing a major release just for this. 
cc: @NipunaRanasinghe

File: ballerina/types.bal (hunk 2 above)
Comment (aashikam) [reply ballerina/types.bal:114]: Screenshot 2024-10-28 at 21 01 45

File: ballerina/modules/soap/records.bal [hunk 4]
```diff
@@ -22,11 +22,61 @@ import ballerinax/'client.config;
...
+    # Configurations associated with retrying
+    http:RetryConfig retryConfig?;
+    # Configurations associated with inbound response size limits
+    http:ResponseLimitConfigs responseLimits?;
+    # SSL/TLS-related options
+    http:ClientSecureSocket secureSocket?;
+    # Proxy server related options
+    http:ProxyConfig proxy?;
+    # Enables the inbound payload validation functionality which provided by the constraint package. Enabled by default
+    boolean validation = true;
```
Comment (NipunaRanasinghe) [inline ballerina/modules/soap/records.bal:56]: Do we need to duplicate all the common configuration under all the submodules? IMO a better approach would be to create a `CommonConfig` record in the default module and, have a type inclusion of that record in all the `ConnectionConfig` types. wdyt?

Comment (aashikam) [reply ballerina/modules/soap/records.bal:56]: Ack, will add a common config to all the clients.

File: ballerina/types.bal (hunk 2 above)
Comment (ayeshLK) [reply ballerina/types.bal:114]: Since we are anyway manually editing this code, do we need to follow the same practice ?

**Comment by ayeshLK** [comment]
Please find my inline comments:

> @ayeshLK
> 
> Used the same client init/configs as same as the Open API tool generated clients: https://github.com/ballerina-platform/module-ballerinax-twilio/blob/17b45c8c2b4d0c846d4f338eedfe6be71abec597/ballerina/types.bal#L25 as requested for the fix.

We are anyway editing this code manually and hence no need to follow the same thing in the generated clients IMO. This just adds new set of records without reusing the existing one's. 

> Major version release is not possible at the moment with how urgent this fix is needed. (There are lot more to be fixed with the next major release: [ballerina-platform/ballerina-library#6697](https://github.com/ballerina-platform/ballerina-library/issues/6697)) Anyway this is up for debate whether or not we are doing a major release just for this. cc: @NipunaRanasinghe

The version should not be determined based on the urgency of a fix or improvement. If a change is breaking, it’s not justified to call it urgent and proceed with it in a patch or minor release. We’ve done this before and have faced consequences as a result. A more effective approach would be to start a discussion to outline the factors we should consider for the next release and establish appropriate versioning for the module.

**Pull request closed without merging**